core/dbt/clients/jinja.py

```diff
@@ -424,3 +468,38 @@ def extract_toplevel_blocks(
         allowed_blocks=allowed_blocks,
         collect_raw_data=collect_raw_data
     )
+
+
+SCHEMA_TEST_KWARGS_NAME = '_dbt_schema_test_kwargs'
+
+
+def add_rendered_test_kwargs(
+    context: Dict[str, Any],
+    node: Union[ParsedSchemaTestNode, CompiledSchemaTestNode],
+    capture_macros: bool = False,
+) -> None:
+    """Render each of the test kwargs in the given context using the native
+    renderer, then insert that value into the given context as the special test
+    keyword arguments member.
+    """
+    looks_like_func = r'^\s*(env_var|ref|var|source|doc)\s*\(.+\)\s*$'
+
+    # we never care about the keypath
+    def _convert_function(value: Any, _: Any) -> Any:
+        if isinstance(value, str):
+            if re.match(looks_like_func, value) is not None:
+                # curly braces to make rendering happy
+                value = f'{{{{ {value} }}}}'
+            new_value = get_rendered(
+                value, context, node, capture_macros=capture_macros,
+                native=True
+            )
+            # this is an ugly way of checking if the value was a quoted str
+            # (we don't want to unquote column names!)
+            if not (len(value) >= 2 and new_value == value[1:-1]):
```

Comment:
I don't think i understand what this check does -- how can `get_rendered` remove quotes around column names?

Reply:
If you take the string `"my_str"` and render it with jinja, jinja will strip the quotes as part of rendering.